apps/class-solid/src/lib/store.ts

```diff
@@ -37,37 +64,105 @@ export async function runExperiment(id: string) {
     }),
   );
 
-  const newOutput = await runClass(exp.config);
+  // TODO make lazy, if config does not change do not rerun
+
+  // Run reference
+  const newOutput = await runClass(exp.reference.config);
+
+  setExperiments(
+    (e) => e.id === exp.id,
+    produce((e) => {
+      e.reference.output = newOutput;
+    }),
+  );
+
+  // Run permutations
+  for (const key in exp.permutations) {
+    const perm = exp.permutations[key];
+    const combinedConfig = mergeConfigurations(
+      exp.reference.config,
+      perm.config,
+    );
+    const newOutput = await runClass(combinedConfig);
```

Comment:
Do I understand correctly that runs are executed synchronously now? Since we await the previous one before starting the next?